### ISSUE TITLE
App doesn't compile the same on Linux as it does on OS X

### ISSUE THREAD

**Issue by mixflame** [Amber-Crystal/amber#125]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem
    * Checked that your issue isn't already filed: https://github.com/Amber-Crystal/amber/issues
    * Checked that there is not already a Crystal Shard that provides the described functionality: http://crystalshards.xyz/

### Description

App compiles and runs perfectly fine on OS X El Capitan 10.11.4

Yet on Linux (tried Debian 8 and Linux Mint) I get:

```
Error in lib/amber/src/amber/router/pipe/static.cr:15: this 'initialize' doesn't initialize instance variable '@directory_listing' of HTTP::StaticFileHandler,
                  with Amber::Pipe::Static < HTTP::StaticFileHandler, rendering it nilable
```

I tried but I can't solve this error. Is this a Crystal compiler bug?

### Steps to Reproduce

1. Write an Amber app
2. Compile and run it on OS X with 0.23.0
3. Try compiling it on Debian 8 64-bit with latest Crystal

**Expected behavior:** App should compile and run like on OS X.

**Actual behavior:** ```amber w```, ```amber r```, and ```crystal run src/myapp.cr``` won't work.

**Reproduces how often:** 100%

### Versions

```Amber CMD (ambercr.io) - v0.1.15```

```Darwin Jonathans-MacBook-Pro-2.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64```

```Linux mixflame 4.9.15-x86_64-linode81 #1 SMP Fri Mar 17 09:47:36 EDT 2017 x86_64 GNU/Linux```

```
Crystal 0.23.0 [3c3d3e2] (2017-06-28) LLVM 3.8.1
```

**Comment by mixflame** [comment]
This is a Crystal 0.23.0 bug. It will happen on OS X as well in 0.23.0.

**Comment by mixflame** [comment]
Fixed by #126
